### PR TITLE
Prekucki/id auto provisioning

### PR DESCRIPTION
**Pull request by prekucki**
# Yagna Server Configuration

## Rationale

The purpose of the change is
 - it enables the activation of a new node based on an etherium key generated in an external one
 configurator.
 - api authorization key generation without the need to start the server
 
## New environmental variables
 
 - `YAGNA_AC_IDENTITY_PK` - private key which activates as a new identity during server startup.
 - `YAGNA_AC_APPKEY` - api key for the application. the key is temporary and may be different each time you run it.

### Case 1

New instalation. Want to run requestor 

```
$ YAGNA_AC_APPKEY=smok1 yagna service run
```

```
$ YAGNA_APPKEY=smok1 my-requestor
```

### Case 2

Paper backup identity key.

```
$ cargo install wagyu
...
...

$ wagyu ethereum hd


      Path                 m/44'/60'/0'/0
      Mnemonic             devote unit reflect material short load round easily combine gadget foot stem
      Private Key          0570e8bc593869fded69dea1ef527d6b63a2cf7f5f557c9996376421075643ad
      Public Key           e1f22f76a25741c9cd4c1e7033c36d8140...96ea35de3da598181554695453026b818dc4d3963
      Address              0x6403a93e6FD70D770Af7b77b97616B7A20Ed874a

$ export YAGNA_AC_IDENTITY_PK=0570e8bc593869fded69dea1ef527d6b63a2cf7f5f557c9996376421075643ad
$ yagna service run

$ yagna id list
{
  "headers": [
    "default",
    "locked",
    "alias",
    "address"
  ],
  "values": [
    [
      "X",
      "",
      null,
      "0x6403a93e6fd70d770af7b77b97616b7a20ed874a"
    ]
  ]
}

```

### CODE REVIEW

**Comment by tworec** [comment]
Great. We've been also considering such a mechanism.
It resolves #1353 and #1534

File: core/identity/src/autoconf.rs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+use ethsign::*;
+use rustc_hex::FromHex;
+
+use ya_core_model::NodeId;
+
+use crate::id_key::IdentityKey;
+use anyhow::Context;
+
+// autoconfiguration
+const ENV_AUTOCONF_PK: &str = "YAGNA_AC_IDENTITY_PK";
```
Comment (tworec) [inline core/identity/src/autoconf.rs:12]: Naming consistency would be better if you use eg.
```suggestion
const ENV_AUTOCONF_PK: &str = "YAGNA_AUTOCONF_ID_PRIV_KEY";
```

File: core/identity/src/autoconf.rs [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+const ENV_AUTOCONF_APP_KEY: &str = "YAGNA_AC_APPKEY";
```
Comment (tworec) [inline core/identity/src/autoconf.rs:13]:
```suggestion
const ENV_AUTOCONF_APP_KEY: &str = "YAGNA_AUTOCONF_APP_KEY";
```

File: core/identity/src/service.rs [hunk 3]
```diff
@@ -1,12 +1,13 @@
-/// Identity service
-use futures::lock::Mutex;
 use std::sync::Arc;
 
-use crate::cli::Command;
+/// Identity service
```
Comment (tworec) [inline core/identity/src/service.rs:3]: this should be first line in the file

**Review by tworec — CHANGES_REQUESTED** [review]

File: core/identity/src/autoconf.rs (hunk 2 above)
Comment (prekucki) [reply core/identity/src/autoconf.rs:13]: Requestors already use `YAGNA_APPKEY`. naming new variable to `YAGNA_AUTOCONF_APP_KEY` would be incosistent.

File: core/identity/src/service/appkey.rs [hunk 4]
```diff
@@ -84,23 +87,53 @@ pub async fn activate(db: &DbExecutor) -> anyhow::Result<()> {
...
+                        .map_err(model::Error::internal)?
+                        .ok_or_else(|| model::Error::internal("appkey not found"))?;
+                    model::AppKey {
+                        name: "autoconfigured".to_string(),
+                        key: get.key.clone(),
+                        role: model::DEFAULT_ROLE.to_string(),
+                        identity: default_identity.node_id,
+                        created_date: start_datetime.clone(),
+                    }
+                })
```
Comment (Wiezzel) [inline core/identity/src/service/appkey.rs:121]:
```suggestion
                let node_id = match preconfigured_node_id {
                    Some(node_id) => node_id,
                    None => {
                        let default_identity = bus::service(idm::BUS_ID)
                            .send(idm::Get::ByDefault)
                            .await
                            .map_err(model::Error::internal)?
                            .map_err(model::Error::internal)?
                            .ok_or_else(|| model::Error::internal("appkey not found"))?;
                        default_identity.node_id
                    }
                };
                Ok(model::AppKey {
                    name: "autoconfigured".to_string(),
                    key: get.key.clone(),
                    role: model::DEFAULT_ROLE.to_string(),
                    identity: node_id,
                    created_date: start_datetime.clone(),
                })
```

Comment (Wiezzel) [inline core/identity/src/service/appkey.rs:121]:
```suggestion
                let node_id = match preconfigured_node_id {
                    Some(node_id) => node_id,
                    None => {
                        let default_identity = bus::service(idm::BUS_ID)
                            .send(idm::Get::ByDefault)
                            .await
                            .map_err(model::Error::internal)?
                            .map_err(model::Error::internal)?
                            .ok_or_else(|| model::Error::internal("appkey not found"))?;
                        default_identity.node_id
                    }
                };
                Ok(model::AppKey {
                    name: "autoconfigured".to_string(),
                    key: get.key.clone(),
                    role: model::DEFAULT_ROLE.to_string(),
                    identity: node_id,
                    created_date: start_datetime.clone(),
                })
```

**Review by Wiezzel — APPROVED** [review]

**Review by tworec — APPROVED** [review]

**Pull request merged**